moduleroot/Rakefile.erb

```
@@ -41,6 +41,11 @@ RSpec::Core::RakeTask.new(:acceptance) do |t|
   t.pattern = 'spec/acceptance'
 end
 
+desc 'Generate REFERENCE.md'
+task :reference do
+  sh 'puppet strings generate --format markdown'
```

should we call this via `bundle exec`? I don't think that everybody has puppet installed in `$PATH` at their workstations. At least I don't.

---

I assume that you'd execute `bundle exec rake reference` then and it'd work that way.

---

Can you make this configurable? In https://github.com/voxpupuli/puppet-extlib#development it's slightly different.

---

what needs to be configurable?

---

> Ah good catch. We also used this in the past on other modules:
> 
> ```
> bundle exec rake strings:generate\[',,,,false,true']
> ```

TIL: there's `require 'puppet-strings/tasks'` which gives you this task. Probably a good addition to puppetlabs_spec_helper to attempt to include this but silently skip it if unavailable.

I'm also thinking about contributing `strings:generate` to https://github.com/puppetlabs/puppet-strings instead of solving it in our modulesync.

---

I think these ideas are great, but not directly impacted by this pull request. I don't see a compelling reason to not merge because one module does something different, when it should be generating the docs like every other module.

---

I still want this to be configurable. We know puppet-extlib already needs customized options and it's trivial to add a variable that would make this possible.

---

I don't get why it needs customizable options.

---

To quote the earlier linked extlib README:

```
Reference documentation is generated using puppet-strings. To regenerate it, please run the rake task as follows.

 % bundle exec rake strings:generate\['lib/puppet/functions/extlib/*.rb,,,,false,true']
```

This is done to avoid generating docs for our deprecated functions.

---

Gotcha.. though if the functions still exist, they should have documentation.

---

To be explicit, I was thinking about `sh 'puppet strings generate --format markdown<%= strings_options %>'` or something like that. Then extlib can add arguments in .sync.yml and we'll have a single command will just work in all modules even if the implementation is slightly different.



> I'm also thinking about contributing `strings:generate` to https://github.com/puppetlabs/puppet-strings instead of solving it in our modulesync.

https://github.com/puppetlabs/puppetlabs_spec_helper/pull/274 (added benefit that PDK users will also get this)

---

https://github.com/puppetlabs/puppet-strings/pull/192 is what I actually meant. It doesn't solve the patterns issue though.